src/Catel.MVVM/Catel.MVVM.Shared/Collections/FastObservableCollection.cs

```diff
@@ -422,6 +443,46 @@ public IDisposable SuspendChangeNotifications(SuspensionMode mode)
         }
 
         /// <summary>
+        /// Synchroniz the list with the suspenstion context.
+        /// </summary>
+        private void SynchronizeFromSuspensionContext()
+        {
+            if (_suspensionContext.NewItems.Count > 0)
+            {
+                _suspensionContext.NewItemIndices.Reverse();
+                _suspensionContext.NewItems.Reverse();
+
+                for (int i = 0; i < _suspensionContext.NewItems.Count; i ++)
+                {
+                    base.InsertItem(_suspensionContext.NewItemIndices[i], _suspensionContext.NewItems[i]);
+                }
+            }
+
+            if (_suspensionContext.OldItems.Count > 0)
+            {
+                SortedList<int, T> sortedList = new SortedList<int, T>();
+                for (int i = 0; i < _suspensionContext.OldItems.Count; i++)
+                {
+                    var value = _suspensionContext.OldItems[i];
+                    var idx = _suspensionContext.OldItemIndices[i];
+                    sortedList.Add(idx, value);
+                }
+
+                _suspensionContext.OldItems.Clear();
```

Comment:
Perhaps instead of clearing create a new list with the initial capacity of the current list count.
With this you can avoid iterating over two list to clear them at the expense of creating two new instances.